### PR TITLE
🥅 Produce warning instead of error for storage base url overrides

### LINKED ISSUE #119593: [gen-l10n] Failed to complete the generation when `FLUTTER_STORAGE_BASE_URL` is overridden
### The issue is breaking every user that is using localizations and a custom `FLUTTER_STORAGE_BASE_URL`.

This also happens with `flutter pub get` if the l10n generation is synthetic.

C 2GWA62)A7VKK 59JJ3%31

The tool will report `Found syntax errors` but not any actual syntax errors.

Related issues:
- #117610
- #119435
- #119585

## Root cause

When `FLUTTER_STORAGE_BASE_URL` is being overridden, the cache getter will try to raise an error to STDERR that the user is using a custom storage base.

https://github.com/flutter/flutter/blob/5b6572f96dd28dfbde8d9a7c3f73a32a45574408/packages/flutter_tools/lib/src/cache.dart#L523-L526

But using an error print will cause the generate tool recognized as an exception.

https://github.com/flutter/flutter/blob/5b6572f96dd28dfbde8d9a7c3f73a32a45574408/packages/flutter_tools/lib/src/localizations/gen_l10n.dart#L1249-L1252

And when it's using synthetic package, `flutter pub get` will also raise exceptions.

https://github.com/flutter/flutter/blob/5b6572f96dd28dfbde8d9a7c3f73a32a45574408/packages/flutter_tools/lib/src/dart/generate_synthetic_packages.dart#L61-L67

## Steps to Reproduce

1. Run `export FLUTTER_STORAGE_BASE_URL=https://storage.flutter-io.cn` in the terminal.
2. `flutter create test_l10n`
3. Setup internationalization for the project according to the [doc](https://docs.flutter.dev/development/accessibility-and-localization/internationalization).
4. `flutter pub get` or `flutter gen-l10n`

flutter doctor -v
```
Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!

[!] Flutter (Channel stable, 3.7.0, on Microsoft Windows [版本 10.0.22000.1455], locale zh-CN)
    • Flutter version 3.7.0 on channel stable at X:\SDK\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b06b8b2710 (7 days ago), 2023-01-23 16:55:55 -0800
    • Engine revision b24591ed32
    • Dart version 2.19.0
    • DevTools version 2.20.1
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn
```

### PR DESCRIPTION
**Pull request by AlexV525**
~Fixes #119593.~

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by AlexV525** [comment]
cc @thkim1011 @christopherfujino

**Comment by christopherfujino** [comment]
Thanks so much for filing the detailed issue and submitting this patch. In an offline discussion with @thkim1011, we determined a more robust solution would be for the gen-l10n sub-system to no longer rely on the state of the logger, so I'm closing this in favor of https://github.com/flutter/flutter/pull/119644.

**Comment by AlexV525** [comment]
The PR may still valid if it aims to change the error log to warning. WDYT? @christopherfujino

**Comment by christopherfujino** [comment]
> The PR may still valid if it aims to change the error log to warning. WDYT? @christopherfujino

Fair enough, that's a good point. Can you delete the test you added to packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart, as after thkim1011's PR lands that will be a no-op.

File: packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart [hunk 1]
```diff
@@ -6,6 +6,7 @@ import 'package:file/memory.dart';
 import 'package:flutter_tools/src/artifacts.dart';
 import 'package:flutter_tools/src/base/file_system.dart';
 import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart:9]: nit. remove extra import

**Review by christopherfujino — APPROVED** [review]
nit about an extra import, otherwise LGTM

File: packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart (hunk 1 above)
Comment (AlexV525) [reply packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart:9]: Sure, I forgot this on the phone.

**Pull request merged**